CHANGES

```diff
@@ -52,6 +52,16 @@ Vernacular Commands
 - Nested proofs may be enabled through the option `Nested Proofs Allowed`.
   By default, they are disabled and produce an error. The deprecation
   warning which used to occur when using nested proofs has been removed.
+- Added option Uniform Inductive Parameters which abstracts over parameters
+  before typechecking constructors, allowing to write for example
+  `Inductive list (A : Type) := nil : list | cons : A -> list -> list.`
```

Comment:
I would much prefer a more backwards compatible version of this option, which simply caused the inductive to error if the parameters were not passed uniformly.  E.g., it would allow
```coq
Inductive list (A : Type) := nil : list A | cons : A -> list A -> list A.
```
And would error on
```coq
Inductive list (A : Type) := nil : list unit | cons : A -> list A -> list A.
```
Is there any reason not to have it behave like this?

Reply:
I think that would be harder, but I'm not sure how much harder. I'll try it and see.
If the option did work this way, how would you use it backwards-compatibly though? Just the line `Set Uniform Inductive Parameters` would error on earlier versions of Coq.

Reply:
Unknown options don't error, they just warn.

Reply:
Well, that makes things a bit simpler.
@JasonGross Check out https://github.com/mit-plv/fiat-crypto/compare/master...jashug:patch_uniform_parameters_2
If all the parameters are implicit, `Set Uniform Inductive Parameters` is pretty much backwards compatible (I made this explicit in `CHANGES`).
I can still look into making the option work as you requested, but my feeling is that treating uniform parameters as though they have been abstracted with a section (so that you can't even mention them in the constructors) is more sensible theoretically and easier to use going forward.

Reply:
That overlay looks much better.  I'd be willing to merge that into fiat-crypto, assuming it is truely backwards compatible.

Reply:
CI on mit-plv/fiat-crypto#383 is all green, so this is indeed backwards compatible.